client/my-sites/site-settings/jetpack-credentials/credentials-setup-flow/index.jsx

```diff
@@ -0,0 +1,139 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Gridicon from 'gridicons';
+import Button from 'components/button';
+import CredentialsForm from '../credentials-form/index';
+
+class CredentialsSetupFlow extends Component {
+	static propTypes = {
+		isPressable: PropTypes.bool,
+		formIsSubmitting: PropTypes.bool,
+		siteId: PropTypes.number,
+		updateCredentials: PropTypes.func,
+		autoConfigCredentials: PropTypes.func,
+		autoConfigStatus: PropTypes.string
+	};
+
+	componentWillMount() {
+		this.setState( { currentStep: 'start' } );
+	}
+
+	reset = () => {
+		this.setState( { currentStep: 'start' } );
+	};
+
+	goToNextStep = () => {
+		switch ( this.state.currentStep ) {
+			case 'start':
+				this.setState( { currentStep: 'tos' } );
+				return;
+			case 'tos':
+				this.setState( { currentStep: 'form' } );
+				return;
+		}
+	};
```

Comment:
Nice! Also fun fact, we have some ways to make this even more declarative - each step has a natural next step so let's just create that mapping…

```js
goToNextStep = () => this.setState( { currentStep: get( {
	start: 'tos',
	tos: 'form',
}, this.state.currentStep, this.state.currentStep ) } ); // default to no state transition
```